ios/Classes/NetIamyellowTiwsWSProxy.m

```diff
@@ -121,8 +124,9 @@ -(void)close:(id)args
 
 -(void)send:(id)msg
 {
-    ENSURE_SINGLE_ARG(msg, NSString);
-
+    // not sure why tianium is passing a __NSArrayM with inside the TiBLob
+    if (![msg isKindOfClass:[NSString class]])
+        msg = [msg[0] data];
```

Comment:
You should have a look at how they do it in the SDK for polymorphic methods. Let me check…

Ok, [this is how they do it](https://github.com/appcelerator/titanium_mobile/blob/818309fbcb2f52f51afae51e346177ce0ff34c02/iphone/Classes/TiUIButton.m#L238) on `TiUIButton#setImage(String/Blob)`.
They use directly the `id` obj and then they pass it to `TiUtils` for further manipulation.

You should ***try*** to use `ENSURE_SINGLE_ARG(msg, id)`, but my obj-c-fu is not so good to predict if `(id*)msg` will throw or not :)

Once you did that you actually have not an array but a concrete object.

My best suggestion is to do the following:

```objc
-(void)send:(id)msg
{
  ENSURE_SINGLE_ARG(args, id);
  if ([msg isKindOfClass:[NSString class]])
  {
    // do something here
  }
  else if ([msg isKindOfClass:[TiBlob class]])
  {
    // have fun here
  }
}
```